packages/debug/src/browser/debug-session.ts

```diff
@@ -119,12 +119,38 @@ export class DebugSessionImpl extends EventEmitter implements DebugSession {
         return this.proceedRequest("threads");
     }
 
-    pause(threadId: number): Promise<DebugProtocol.PauseResponse> {
+    pause(threadId?: number): Promise<DebugProtocol.PauseResponse> | Promise<DebugProtocol.PauseResponse[]> {
+        if (!threadId) {
+            return this.threads().then(response => {
+                const threads: DebugProtocol.Thread[] = response.body.threads;
+                const requests = [];
+                for (const thread of threads) {
+                    const pauseResponse = this.proceedRequest("pause", { threadId });
+                    this.emit('changeStatus', thread.id, false);
+                    requests.push(pauseResponse);
+                }
+                return Promise.all(requests);
+            });
+        }
+        this.emit('changeStatus', threadId, false);
```

Comment:
`StoppedEvent` is fired when thread is stopped. You don't have emit `changeStatus` event.

Reply:
Don't emit any events here.
Just rely on `DebugState` https://github.com/theia-ide/theia/blob/vscode-debug-protocol/packages/debug/src/browser/debug-session.ts#L27

Let me know if you need something else.